WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStatusService.kt

```diff
@@ -72,8 +76,9 @@ class ScanStatusService
         return false
     }
 
-    private fun updateScanState(scanState: ScanStateModel?) {
-        _scanAvailable.value = scanState?.state != UNAVAILABLE && scanState?.state != UNKNOWN
+    private fun updateScanState(scanStateModel: ScanStateModel) {
+        _scanAvailable.value = scanStateModel.state != UNAVAILABLE && scanStateModel.state != UNKNOWN
+        _scanState.value = scanStateModel
```

Comment:
Question (❓ ): While reviewing the code, and since I haven't seen this class before, trying to answer the `Flow` question that you posted, I have an extra question about the `reloadScanState`. So, when the `start(...)` function get triggered, wouldn't `reloadScanState()` be triggered twice? Once during `requestScanStateUpdate()` and once directly? Why do we want that?

Reply:
I think this is intended. We want to load the screen with the last state and then reload it again after the fetch is complete.  Do you see any downside?

Reply:
Ah ok, that makes sense! 🙏 No, no major downside I can think of, I would have to think about it a bit more to find an alternative (if any).